### PR TITLE
Automated GitHub build restructure

### PR DESCRIPTION
**Pull request by adelosa**
The following PR is to replace the existing automated build for the project.

The issues with the current build:
* Tests are completely run twice - for Linux and Windows. Given it is Java (write once, run anywhere) a full regression is not necessary for each platform.
* Jar is recreated multiple times. Binary included in distribution is not actually tested
* Distribution has work files that are generated as part of the test process.
* Current PDF formatting is not fantastic - removing its inclusion until I find something better

New build takes the following approach:

Workflow: build.yaml
* Job 1:  Build z390 distribution zip file by using bash/blddist script - save the Distribution zip
* Job 2: Run full regression tests on Linux using distribution zip from job 1
* Job 3: Run ASM and Cobol demo scripts on Windows using distribution zip from job 1
* Job 4: If Tag - create a GitHub release using the distribution zip from job 1

Workflow: docs.yaml
* Job 1: Build HTML docs from project doc folder and publish to GitHub pages (no changes except updates to py version)

### CODE REVIEW

**Comment by adelosa** [comment]
*** Need to fix tests before merge ***

The gradle tests are run against the jar and source in the distribution zip.
So build gets zip, extracts the content and then runs the tests against that source.
This causes the folder name to the source to be longer which causes the source to not assemble/compile
because the line is too long.

There are 3 such tests in the testbase.
* RunCblDemos.groovy -> zCOBOL/demo/COPYFILE
* RunCmdProc.groovy -> rt/test/TESTCMD3
* RunCmdProc.groovy -> rt/test/TESTCMD4

These tests need to be updated to remove the filenames and use files referenced in envvars instead.

------

Make sure to squash before merging as there are lots of work commits in this pull request as testing involves pushing to GitHub to run the actions.

I fixed an issue with the javadoc build `bldjavadoc`- it was placing its output in the `docenv` folder which should only be used for the python virtual environment that the doc build uses - I have changed it to use `build/javadoc` folder instead.
I also added a param for `bldjavadoc` to provide an optional output folder in case you don't want to use `build/javadoc`. 
I also removed the execution of `bldjavadoc` from `bldjar` and I now execute it from `blddist` instead. 

If you use the javadocs, then you will need to rerun bldjavadoc after bldjar. I have not checked the BAT jobs for consistency.. @abekornelis - your thoughts would be appreciated as I think you added the javadoc build into the bldjar function. For the build, it makes things easier not having multiple levels of scripts.

**Comment by abekornelis** [comment]
Anthony, I did indeed add the javadoc gen step to the jar build. John was not too happy with the details of the implementation, if I remember correctly. In hindsight I've had my own misgivings. I'm entirely fine if you take it out again. The javadoc can still be built whenever needed. As part of the build process would be required. Whether or not for a distro. A build should build the full product including all bells and whistles. Fine with me if you change it. Or if you create an issue to make the change at some later time.

**Comment by adelosa** [comment]
I have tested this now with @jyganci changes to remove the hard coded filenames. Build is now working. 
Ready for review and merge if OK.

File: bash/blddist
Comment (jyganci) [inline bash/blddist]: Question only. Is this line here (also line 18 of build.yml) because of issue 518 zc390.java fails when directory name contains a period? If yes, a potential fix exists. See issue 518 comment added Oct 7.

File: bash/blddist
Comment (jyganci) [inline bash/blddist]: Remove the "# unix bash scripts" comment.

File: bash/blddist
Comment (jyganci) [inline bash/blddist]: Remove the "# unix bash scripts" comment.

File: bash/blddist
Comment (jyganci) [inline bash/blddist]: Remove the "# unix bash scripts" comment.

**Review by jyganci — COMMENTED** [review]
I'm ready to approve after you've responded to comments added to the PR.

File: bash/blddist
Comment (adelosa) [reply bash/blddist]: I have added your fix and removed the workaround.

File: bash/blddist
Comment (adelosa) [reply bash/blddist]: Removed the comment

File: bash/blddist
Comment (adelosa) [reply bash/blddist]: Removed

File: bash/blddist
Comment (adelosa) [reply bash/blddist]: Removed

**Comment by adelosa** [comment]
Hi @jyganci - I have updated according to your comments and incorporated the fix for the dot in directory name.

**Comment by abekornelis** [comment]
I am fine with these changes. I will let John approve the PR as you both have spent considerable more time and effort than I did.

File: .github/workflows/build.yml
Comment (jyganci) [inline .github/workflows/build.yml:52]: Question only. This looks like 2 statements: set Z390_PROJECT_ROOT and run z390test/gradlew. You can have 2 statements on the same line? I'm probably missing something.

**Review by jyganci — APPROVED** [review]
Looks good. I did ask one question. Answer that and merge the PR.

File: .github/workflows/build.yml
Comment (adelosa) [reply .github/workflows/build.yml:52]: 2 statements. One sets the envvar then runs the command. I believe the envvar scope is just for that command. I see it as an extension of the command.

**Comment by jyganci** [comment]
Interesting. Learned something new.

**Pull request merged**
